gui/wxpython/lmgr/frame.py

```diff
@@ -1115,28 +1064,27 @@ def ChangeLocation(self, location, mapset, dbase=None):
 
     def OnCreateMapset(self, event):
         """Create new mapset"""
-        dlg = wx.TextEntryDialog(parent=self,
-                                 message=_('Enter name for new mapset:'),
-                                 caption=_('Create new mapset'))
-
-        if dlg.ShowModal() == wx.ID_OK:
-            mapset = dlg.GetValue()
-            if not mapset:
-                GError(parent=self,
-                       message=_("No mapset provided. Operation canceled."))
-                return
+        db_node, loc_node, mapset_node = self.datacatalog.tree.GetCurrentDbLocationMapsetNode()
+        mapset = create_mapset_interactively(self, db_node.data['name'],
+                                             loc_node.data['name'])
+        if mapset:
+            self.datacatalog.tree.InsertMapset(name=mapset,
+                              location_node=loc_node)
+        self.datacatalog.tree.ReloadTreeItems()
 
-            ret = RunCommand('g.mapset',
-                             parent=self,
-                             flags='c',
-                             mapset=mapset)
-            # ensure that DB connection is defined
-            ret += RunCommand('db.connect',
-                              parent=self,
-                              flags='c')
-            if ret == 0:
-                GMessage(parent=self,
-                         message=_("Current mapset is <%s>.") % mapset)
+    def OnLocationWizard(self, event):
+        """Create new location"""
+        grassdb_node, loc_node, mapset_node = self.datacatalog.tree.GetCurrentDbLocationMapsetNode()
+        grassdatabase, location, mapset = (
+            create_location_interactively(self, grassdb_node.data['name'])
+        )
+        if location:
+            grassdb_nodes = self.datacatalog.tree._model.SearchNodes(name=grassdatabase,
+                                                                     type='grassdb')
+            if not grassdb_nodes:
+                grassdb_node = self.datacatalog.tree.InsertGrassDb(name=grassdatabase)
+            self.datacatalog.tree.InsertLocation(location, grassdb_node)
+        self.datacatalog.tree.ReloadTreeItems()
```

Comment:
```suggestion
        if location:
            self.datacatalog.tree.ReloadTreeItems()
```
For now, let's ignore the case when location is created in a db which is not in data catalog. We would have to implement that differently anyway, SearchNodes and InsertGrassDb would have to be at least wrapped in catalog.py. This will be improved later.